File: tenis-quarkus/src/main/java/br/unitins/joaovittor/basqueteiros/Material/service/MaterialService.java
package br.unitins.joaovittor.basqueteiros.Material.service;

import java.util.List;
import br.unitins.joaovittor.basqueteiros.Material.dto.MaterialDTO;
import br.unitins.joaovittor.basqueteiros.Material.dto.MaterialResponseDTO;
import jakarta.validation.Valid;

public interface MaterialService {

    public MaterialResponseDTO create(@Valid MaterialDTO dto);
    public void update(Long id, MaterialDTO dto);
    public boolean delete(Long id);
    public List<MaterialResponseDTO> findAll(int page, int pageSize);
    public List<MaterialResponseDTO> findAll();
    public MaterialResponseDTO findById(Long id);
    public List<MaterialResponseDTO> findByDescricao(String descricao);
    public List<MaterialResponseDTO> findByDescricao(int page, int pageSize, String descricao);
    public MaterialResponseDTO findByCategoria(String categoria);
    public long count();

} 

File: tenis-angular/src/app/services/material.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Material } from '../models/material.model';

@Injectable({
  providedIn: 'root'
})
export class MaterialService {
  private baseUrl = 'http://localhost:8080/materiais';

  constructor(private httpClient : HttpClient){

  }
  findAll(): Observable<Material[]>;
  findAll(page: number, pageSize: number): Observable<Material[]>;

  findAll(page?: number, pageSize?: number): Observable<Material[]> {
    let params = {};
    if (page !== undefined && pageSize !== undefined) {
      params = {
        page: page.toString(),
        pageSize: pageSize.toString()
      }
    }
    // console.log(params);
    return this.httpClient.get<Material[]>(this.baseUrl, {params}); 
  }
  count(): Observable<number> {
    return this.httpClient.get<number>(`${this.baseUrl}/count`); 
  }

  findById(id: string): Observable<Material> {
    return this.httpClient.get<Material>(`${this.baseUrl}/search/id/${id}`); 
  }

  insert(material: Material): Observable<Material> {
    return this.httpClient.post<Material>(this.baseUrl, material);
  }

  update(material: Material): Observable<Material> {
    const data = {
      descricao: material.descricao,
      categoria: material.categoria
    }
    return this.httpClient.put<any>(`${this.baseUrl}/${material.id}`, data); 
  }

  delete(material: Material): Observable<any>{
    return this.httpClient.delete<any>(`${this.baseUrl}/${material.id}`); 
  }
}
